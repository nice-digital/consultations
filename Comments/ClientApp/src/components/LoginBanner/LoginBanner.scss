.sign-in-banner {
  margin-top: 0.5rem !important;
}

.panel-white {
  background-color: #fff;
}

.display--inline {
  display: inline;
}

.no-margin {
  margin: 0;
}
