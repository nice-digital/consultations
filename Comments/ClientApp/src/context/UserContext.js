// @flow

import React from "react";
import Cookies from "js-cookie";
import { withRouter } from "react-router";

import { load } from "../data/loader";
import preload from "../data/pre-loader";

export const UserContext = React.createContext({
	updateContext: () => {},
});

type PropsType = {
	location: any,
	children: any,
	staticContext: any,
	match: any,
};

type StateType = {
	isAuthorised: boolean,
	isOrganisationCommenter: boolean,
	isLead: boolean,
	displayName: string,
	signInURL: string,
	registerURL: string,
	organisationName: string,
	initialDataLoaded: boolean,
};

export class UserProvider extends React.Component<PropsType, StateType> {
	constructor(props: PropsType) {
		super(props);

		this.state = {
			isAuthorised: false,
			isOrganisationCommenter: false,
			isLead: false,
			displayName: "",
			signInURL: "",
			registerURL: "",
			updateContext: this.updateContext,
			organisationName: null,
			initialDataLoaded: false,
		};

		const isServerSideRender = (this.props.staticContext && this.props.staticContext.preload);
		const preloadSource = isServerSideRender ? this.props.staticContext.preload.data : window.__PRELOADED__; // TODO: extract this preloaded line out to (or near) the preload endpoint method.

		const userSessionParameters = this.getUserSessionParameters(preloadSource);

		let isOrganisationCommenter = false;
		let organisationName = null;
		let isAuthorised = preloadSource ? preloadSource.isAuthorised : false;

		if (userSessionParameters.sessionCookieExistsForThisConsultation){
			const preloadedUserSessionData = preload(
				this.props.staticContext,
				"checkorganisationusersession",
				[],
				{
					consultationId: userSessionParameters.consultationId,
				},
				preloadSource,
			);

			if (preloadedUserSessionData){
				isOrganisationCommenter = preloadedUserSessionData.valid;
				isAuthorised = isAuthorised || preloadedUserSessionData.valid ; //you could be authorised by idam or by organisation session cookie.
				organisationName = preloadedUserSessionData.organisationName;
			}
		}

		if (preloadSource){
			this.state = {
				isAuthorised,
				isOrganisationCommenter: isOrganisationCommenter,
				isLead: preloadSource.isLead,
				displayName: preloadSource.displayName,
				signInURL: preloadSource.signInURL,
				registerURL: preloadSource.registerURL,
				updateContext: this.updateContext,
				organisationName: organisationName,
				initialDataLoaded: true,
			};
			if (this.props.staticContext) {
				this.props.staticContext.analyticsGlobals.isSignedIn = preloadSource.isAuthorised;
			}
		}
	}

	setStateForValidSessionCookie = (organisationName) => {
		this.setState({
			isAuthorised: true,
			isOrganisationCommenter: true,
			organisationName,
		});
	}

	loadUser = (returnURL) => {
		load("user", undefined, [], { returnURL, cachebust: new Date().getTime() })
			.then(
				res => {
					const signInURL = res.data.signInURL;
					this.setState({
						isAuthorised: res.data.isAuthenticatedByAccounts,
						displayName: res.data.displayName,
						signInURL: signInURL,
						registerURL: res.data.registerURL,
						isLead: (res.data.organisationsAssignedAsLead && res.data.organisationsAssignedAsLead.length > 0),
					});
					//update signin links in global nav here. because SSR isn't rendering them right on the server.
					var signInLinks = document.getElementById("global-nav-header").querySelectorAll("a[href*='account/login']");
					for (var i=0; i < signInLinks.length; i++) {
						signInLinks[i].setAttribute("href", signInURL);
					}
				},
			)
			.then(
				this.checkSessionId()
					.then(data => {
						if (data.validityAndOrganisationName?.valid === true) {
							this.setStateForValidSessionCookie(data.validityAndOrganisationName.organisationName);
						}
					}),
			);
	}

	getFeatureFlags = async () => {
		load("getfeatureflags", undefined, [], {})
			.then(response =>
				this.setState({
					organisationalCommentingFeature: response.data.OrganisationalCommenting,
				}))
			.catch(err => {
				console.log(JSON.stringify(err));
			});
	}

	//unfortunately the context is above the routes, so this.props.match is always null, so we can't pull the consultation id out of there. hence we're falling back to regex.
	getConsultationId = () =>{
		const regex = /^\/(\d+)\/(review|submitted|\d+\/[a-z-A-Z0-9]+)/;
		const pathname = this.props.location.pathname;

		if (!pathname)
			return;

		const matches = regex.exec(pathname);
		if (!matches || matches.length !== 3)
			return;

		return matches[1];
	}

	checkSessionId = async () => {
		const userSessionParameters = this.getUserSessionParameters(window.__PRELOADED__);

		if (!userSessionParameters.sessionCookieExistsForThisConsultation)
			return await {
				validityAndOrganisationName: {valid: false},
				userSessionParameters: userSessionParameters,
			};

		const validityAndOrganisationName = load(
			"checkorganisationusersession",
			undefined,
			[],
			{
				consultationId: userSessionParameters.consultationId,
			})
			.then(response => response.data)
			.catch(err => {
				console.log(JSON.stringify(err));
				return {valid: false};
			});
		return {
			validityAndOrganisationName: await validityAndOrganisationName,
			userSessionParameters: userSessionParameters,
		};
	}

	getUserSessionParameters = (preloadSource) => {
		const consultationId = this.getConsultationId();
		const cookieName = `ConsultationSession-${consultationId}`;
		const cookieString = preloadSource?.cookies || "";
		const sessionCookieExistsForThisConsultation =  (Cookies.get(cookieName) || cookieString.indexOf(cookieName) !== -1);
		return {consultationId, sessionCookieExistsForThisConsultation};
	}

	//a child component can call this method to update the context in case a session cookie has been set.
	updateContext = () => {
		this.checkSessionId()
			.then(data => {
				if (data.validityAndOrganisationName.valid === true) {
					this.setStateForValidSessionCookie(data.validityAndOrganisationName.organisationName);
				}
			});
	}

	loadContext = () => {
		this.loadUser(this.props.location.pathname);
	}

	componentDidMount() {
		this.loadContext(); //this is needed here as the sign in url isn't right on SSR. TODO: fix SSR.
	}

	componentDidUpdate(prevProps) {
		if (this.props.location !== prevProps.location) { //fired when the route changes, as you might be authenticated for some routes, but not others (e.g. org commenting cookie)
			this.loadContext();
		}
	}

	render() {
		return (
			<UserContext.Provider value={this.state}>
				{this.props.children}
			</UserContext.Provider>
		);
	}
}

export default withRouter(UserProvider);
