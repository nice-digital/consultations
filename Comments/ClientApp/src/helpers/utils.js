// @flow

import queryString from "query-string";
//import stringifyObject from "stringify-object";

/**
 * Create a query string out of an object
 * @param obj the key-value pairs that you'd like to be turned into a querystring
 * @returns {string}
 */
export function objectToQueryString(obj) {
	let str = [];
	for (let p in obj)
		if (Object.prototype.hasOwnProperty.call(obj, p)) {
			if (Array.isArray(obj[p])) {
				for (let index in obj[p]) {
					str.push(encodeURIComponent(p) + "=" + encodeURIComponent(obj[p][index]));
				}
			} else {
				str.push(encodeURIComponent(p) + "=" + encodeURIComponent(obj[p]));
			}
		}
	if (str.length) return "?" + str.join("&");
	return "";
}

/**
 * Creates an object out of a query string
 * 	@param qryString {string} the querystring you'd like to be turned into an object
 *  @returns obj
 */
export function queryStringToObject(qryString) {
	return queryString.parse(qryString);
}

/**
 * Resolve a promise on the next tick - used in testing
 * @see https://nodejs.org/en/docs/guides/event-loop-timers-and-nexttick/#process-nexttick
 * @returns {Promise<any>}
 */
export const nextTick = async () => {
	return new Promise((resolve) => {
		setTimeout(resolve, 0);
	});
};

/**
 * Format a string by replacing {0} with arguments passed in
 * @see https://stackoverflow.com/a/4673436
 * @returns {String}
 */
export function replaceFormat(stringToReplace, args) {
	if (typeof (stringToReplace) === "undefined" || stringToReplace === null)
		return stringToReplace;

	return stringToReplace.replace(/{(\d+)}/g, function (match, number) {
		return typeof args[number] !== "undefined"
			? args[number]
			: match
		;
	});
}

/**
 * @param link String The string in which you want to check for the presence of "http"
 * Detects link that starts with "http"
 * @returns {Boolean}
 */
export const isHttpLink = link => link.indexOf("http") === 0;

// Remove querystring parameters from a given URL
export const removeQueryParameter =
	(url: string, parameter: string, value: ?string = null): string => {
		const urlParts = url.split("?");

		if (urlParts.length < 2)
			return url;

		let urlParams = urlParts[1].split(/[&;]/g);

		for (let i = urlParams.length; i-- > 0;) {
			const parParts = urlParams[i].split("=");

			if (parParts[0].toLowerCase() === parameter.toLowerCase()
				&& (!value || (value && value.toLowerCase() === (parParts[1] || "").toLowerCase()))) {
				urlParams.splice(i, 1);
			}
		}

		const queryString = (urlParams.length > 0 ? "?" + urlParams.join("&") : "");
		return urlParts[0] + queryString;
	};

export const stripMultipleQueries = (path, queries) => {
	let strippedPath = path;

	queries.forEach((query) => {
		strippedPath = removeQueryParameter(strippedPath, query);
	});

	return strippedPath;
};

export const appendQueryParameter =
	(url: string, parameter: string, value: string): string => `${url}${url.indexOf("?") === -1 ? "?" : "&"}${parameter}=${value}`;

/// Determines if the execution environment is client or server
/// See https://stackoverflow.com/a/32598826/486434
export const canUseDOM = (): boolean => (
	!!(
		typeof window !== "undefined" &&
		window.document &&
		window.document.createElement
	));

const whichChild = (elem) => {
	let i = 0;
	while ((elem == elem.previousSibling) != null) ++i;
	return i;
};

const reverseString = (str) => {
	return (str === "") ? "" : reverseString(str.substr(1)) + str.charAt(0);
};

//this function returns a long dotted decimal representing the position in the document. below the div with id of root.
export const getElementPositionWithinDocument = (elem) => {
	let curindex = "";
	if (elem.parentElement) {
		do {
			curindex += whichChild(elem).toString() + " "; // add elem.tagName to see what the element is.
			if (elem.id === "root") {
				break;
			}
		} while (elem = elem.parentElement); // eslint-disable-line
	}
	return reverseString(curindex.trim().replace(new RegExp(/\s/, "g"), "."));
};

//this gets the title of the nearest section above the passed in element (or the element itself). it can also return the chapter title.
export const getSectionTitle = (elem) => {
	do {
		if (elem.classList.contains("section")) {
			return elem.title;
		}
		if (elem.classList.contains("chapter")) {
			return elem.title;
		}
		if (elem.id === "root") { //it won't look higher than root.
			return null;
		}
	} while (elem = elem.parentNode); // eslint-disable-line
	return null; //shouldn't really ever be called, as root will get hit if there's no matches.
};

export const removeQuerystring = (url) => {
	return url.split("?")[0];
};
