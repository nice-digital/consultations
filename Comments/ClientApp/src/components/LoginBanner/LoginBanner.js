// @flow
import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import { DebounceInput } from "react-debounce-input";
import queryString from "query-string";
import Cookies from "js-cookie";
import { UserContext } from "../../context/UserContext";
import { load } from "../../data/loader";
import { Input } from "@nice-digital/nds-input";

type PropsType = {
	signInURL: string,
	registerURL: string,
	signInButton: boolean,
	signInText?: string,
	match: PropTypes.object.isRequired,
	location: PropTypes.object.isRequired,
	allowOrganisationCodeLogin: boolean,
	orgFieldName: string, //the organisation text input needs a unique name, however the login banner will be on the page twice. so passing in a unique name
	codeLoginOnly: boolean,
	title: string,
	isInCommentsPanel: boolean,
}

type OrganisationCode = {
	organisationAuthorisationId: number,
	organisationId: number,
	organisationName: string,
	collationCode: string,
}

type StateType = {
	userEnteredCollationCode: string,
	hasError: boolean,
	errorMessage: string,
	showAuthorisationOrganisation: boolean,
	authorisationOrganisationFound: OrganisationCode,
}

export class LoginBanner extends Component<PropsType, StateType> {

	constructor(props: PropsType) {
		super(props);

		this.state = {
			userEnteredCollationCode: "",
			hasError: false,
			errorMessage: "",
			showAuthorisationOrganisation: false,
			authorisationOrganisationFound: null,
		};
	}

	componentDidMount() {
		const userEnteredCollationCode = queryString.parse(this.props.location.search).code;
		if (userEnteredCollationCode){
			this.setState({
				userEnteredCollationCode,
			}, () => {
				this.checkOrganisationCode();
			});
		}
	}

	handleOrganisationCodeChange = (userEnteredCollationCode) => {
		this.setState({
			userEnteredCollationCode,
		}, () => {
			if (userEnteredCollationCode){ //if there's a blank value, don't bother hitting the server.
				this.checkOrganisationCode(userEnteredCollationCode);
			}else{
				this.setState({
					hasError: false,
					showAuthorisationOrganisation: false,
				});
			}
		});
	};

	gatherDataForCheckOrganisationCode = async () => {
		const organisationCode = load(
			"organisation",
			undefined,
			[],
			{
				collationCode: this.state.userEnteredCollationCode,
				consultationId: this.props.match.params.consultationId,
			})
			.then(response => response.data)
			.catch(err => {
				this.setState({
					hasError: true,
					errorMessage: err.response.data.errorException.Message,
					showAuthorisationOrganisation: false,
				});
			});
		return {
			organisationCode: await organisationCode,
		};
	}

	checkOrganisationCode = () => {
		this.gatherDataForCheckOrganisationCode()
			.then(data => {
				if (data.organisationCode != null) {
					this.setState({
						hasError: false,
						errorMessage: "",
						showAuthorisationOrganisation: true,
						authorisationOrganisationFound: data.organisationCode,
					});
				}
			})
			.catch(err => {
				throw new Error("checkOrganisationCode failed " + err);
			});
	}

	gatherDataForCreateOrganisationUserSession = async () => {
		const session = load(
			"organisationsession",
			undefined,
			[],
			{
				collationCode: this.state.userEnteredCollationCode,
				organisationAuthorisationId: this.state.authorisationOrganisationFound.organisationAuthorisationId,
			},
			"POST")
			.then(response => response.data)
			.catch(err => {
				this.setState({
					hasError: true,
					errorMessage: err.response.data.errorException.Message,
					showAuthorisationOrganisation: false,
				});
			});
		return {
			session: await session,
		};
	}

	createOrganisationUserSession = async () => {
		const session = this.gatherDataForCreateOrganisationUserSession()
			.then(data => {
				if (data.session != null) {
					this.setState({
						hasError: false,
						errorMessage: "",
					});
					return data.session;
				}
			})
			.catch(err => {
				throw new Error("checkOrganisationCode failed " + err);
			});
		return {
			session: await session,
		};
	}

	handleConfirmClick = (updateContextFunction) => {
		const consultationId = this.props.match.params.consultationId;
		this.createOrganisationUserSession().then(data => {
			console.log("data is:" + JSON.stringify(data));

			var expirationDate = new Date(data.session.expirationDateTicks);

			Cookies.set(`ConsultationSession-${consultationId}`, data.session.sessionId, {expires: expirationDate});

			//now, set state to show logged in.
			this.setState({
				showAuthorisationOrganisation: false,
			});
			updateContextFunction();
		})
			.catch(() => {
				this.setState({
					hasError: true,
					errorMessage: "Unable to confirm",
				});
			});
	}

	render(){
		const limitWidthOfButton = !this.props.signInButton; //the sign-in button isn't shown when we're trying to save space.

		const codeLoginOnly = this.props.codeLoginOnly ?? false;
		const title = this.props.title ?? "";
		const isInCommentsPanel = this.props.isInCommentsPanel ?? false;

		return (
			<div className={`${!isInCommentsPanel ? "panel panel-white" : ""} mt--0 mb--0 sign-in-banner`} id="loginBanner" data-qa-sel="sign-in-banner">
				<div className="container">
					<div className="LoginBanner" role="form">
						{title !== "" &&
							<h3>{title}</h3>
						}
						{this.props.allowOrganisationCodeLogin &&
							<>
								<div className={this.state.hasError ? "input input--error" : "input"}>
									<DebounceInput
										minLength={5}
										debounceTimeout={400}
										type="text"
										onChange={e => this.handleOrganisationCodeChange(e.target.value)}
										className={limitWidthOfButton ? "limitWidth input__input" : "input__input"}
										data-qa-sel="OrganisationCodeLogin"
										value={this.state.userEnteredCollationCode}
										element={Input}
										error={this.state.hasError}
										errorMessage={this.state.errorMessage}
										label="Enter your organisation code"
										name={"orgCode-" + this.props.orgFieldName}
									/>
								</div>
								{this.state.showAuthorisationOrganisation &&
									<>
										<p>Confirm organisation name</p>
										<p><strong>{this.state.authorisationOrganisationFound.organisationName}</strong></p>
										<UserContext.Consumer>
											{/* eslint-disable-next-line*/}
											{({ contextValue: ContextType, updateContext }) => (
												<div>
													<button className="btn btn--cta" data-qa-sel="ConfirmOrgNameButton" onClick={() => this.handleConfirmClick(updateContext)}  title={"Confirm your organisation is " + this.state.authorisationOrganisationFound.organisationName}>Confirm</button>
												</div>
											)}
										</UserContext.Consumer>
									</>
								}
							</>
						}
						{!codeLoginOnly &&
							<>
								{this.props.allowOrganisationCodeLogin && this.props.signInButton &&
									<>
										<p>If you don't have an organisation code, sign in to your NICE account.</p>
										<p>
											<a className="btn" href={this.props.signInURL} title="Sign in to your NICE account">Sign in</a>
										</p>
									</>
								}
								{this.props.allowOrganisationCodeLogin && !this.props.signInButton &&
									<>If you don't have an organisation code, <a href={this.props.signInURL} title="Sign in to your NICE account">sign in to your NICE account.</a>&nbsp;&nbsp;</>
								}
							</>
						}
						{!this.props.allowOrganisationCodeLogin &&
							<>
								<p className={`${!isInCommentsPanel ? "display--inline" : ""}`}><a href={this.props.signInURL} title="Sign in to your NICE account">Sign in to your NICE account</a> {this.props.signInText ?? "to comment on this consultation"}.{" "}</p>
								{this.props.signInButton &&
									<p>
										<a className="btn" href={this.props.signInURL} title="Sign in to your NICE account">Sign in</a>
									</p>
								}
							</>
						}
						{!codeLoginOnly &&
							<p className={`${!isInCommentsPanel ? "display--inline" : ""}`}>
								<a href={this.props.registerURL} title="Register for a NICE account">
									Register
								</a>
								{" "}
								for a NICE account if you don't already have one.
							</p>
						}
					</div>
				</div>
			</div>
		);
	}
}

export default withRouter(LoginBanner);
LoginBanner.contextType = UserContext;
