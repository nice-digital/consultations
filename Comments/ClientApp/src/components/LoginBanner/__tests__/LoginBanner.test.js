/* eslint-env jest */

import React from "react";
import { mount } from "enzyme";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import toJson from "enzyme-to-json";

import { nextTick } from "../../../helpers/utils";
import OrganisationCode from "./OrganisationCode.json";
import { UserContext } from "../../../context/UserContext";
import { LoginBanner } from "../LoginBanner";

const mock = new MockAdapter(axios);

describe("[ClientApp] ", () => {
	describe("LoginBanner Component", () => {
		const fakeProps = {
			match: {
				url: "/1/1/introduction",
				params: {
					consultationId: 1,
				},
			},
			location: {
				pathname: "/1/1/introduction",
				search: "",
			},
			history:{
				location:{
					search: "",
				},
				listen: function(){},
			},
			signInURL: "/signin-url",
			registerURL: "/register-url",
			signInButton: true,
			signInText: "signin text",
		};

		let contextWrapper = null;

		beforeEach(() => {
			mock
				.onGet()
				.reply(200, OrganisationCode);

			contextWrapper = {
				wrappingComponent: UserContext.Provider,
				wrappingComponentProps: {
					value: { isAuthorised: true },
				},
			};
		});

		afterEach(() => {
			mock.reset();
		});

		it("should match snapshot for idam login and not in comments panel", async () => {
			const wrapper = mount(
				<LoginBanner {...fakeProps} />,
				contextWrapper,
			);

			await nextTick();
			wrapper.update();

			expect(toJson(wrapper, {noKey: true, mode: "deep"})).toMatchSnapshot();

			wrapper.unmount();
		});

		it("should match snapshot for idam login and in comments panel", async () => {
			fakeProps.isInCommentsPanel = true;
			fakeProps.title = "Some title";
			fakeProps.signInButton = false;
			fakeProps.signInText = null;

			const wrapper = mount(
				<LoginBanner {...fakeProps} />,
				contextWrapper,
			);

			await nextTick();
			wrapper.update();

			expect(toJson(wrapper, {noKey: true, mode: "deep"})).toMatchSnapshot();

			wrapper.unmount();
		});

		it("should match snapshot for code login only", async () => {
			fakeProps.allowOrganisationCodeLogin = true;
			fakeProps.codeLoginOnly = true;
			fakeProps.title = null;

			const wrapper = mount(
				<LoginBanner {...fakeProps} />,
				contextWrapper,
			);

			await nextTick();
			wrapper.update();

			expect(toJson(wrapper, {noKey: true, mode: "deep"})).toMatchSnapshot();

			wrapper.unmount();
		});

		it("organisation code should be prepopulated from querystring", async () => {
			const expectedOrganisationCode = "123412341234";
			const updatedProps = Object.assign(fakeProps, {location: {pathname: "/1/1/introduction", search:`?code=${expectedOrganisationCode}`}});

			const wrapper = mount(
				<LoginBanner {...updatedProps} />,
				contextWrapper,
			);

			await nextTick();
			wrapper.update();

			expect(wrapper.state().userEnteredCollationCode).toEqual(expectedOrganisationCode);

			wrapper.unmount();
		});

	});
});
