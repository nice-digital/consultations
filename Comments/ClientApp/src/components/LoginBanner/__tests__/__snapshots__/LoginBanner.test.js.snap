// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ClientApp]  LoginBanner Component should match snapshot for code login only 1`] = `
<div
  className=" mt--0 mb--0 sign-in-banner"
  data-qa-sel="sign-in-banner"
  id="loginBanner"
>
  <div
    className="container"
  >
    <div
      className="LoginBanner"
      role="form"
    >
      <div
        className="input"
      >
        <div
          className="input"
        >
          <label
            className="input__label"
            htmlFor="orgCode-undefined"
          >
            Enter your organisation code
          </label>
          <input
            className="limitWidth input__input"
            data-qa-sel="OrganisationCodeLogin"
            id="orgCode-undefined"
            name="orgCode-undefined"
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[ClientApp]  LoginBanner Component should match snapshot for idam login and in comments panel 1`] = `
<div
  className=" mt--0 mb--0 sign-in-banner"
  data-qa-sel="sign-in-banner"
  id="loginBanner"
>
  <div
    className="container"
  >
    <div
      className="LoginBanner"
      role="form"
    >
      <h3>
        Some title
      </h3>
      <p
        className=""
      >
        <a
          href="/signin-url"
          title="Sign in to your NICE account"
        >
          Sign in to your NICE account
        </a>
         
        to comment on this consultation
        .
         
      </p>
      <p
        className=""
      >
        <a
          href="/register-url"
          title="Register for a NICE account"
        >
          Register
        </a>
         
        for a NICE account if you don't already have one.
      </p>
    </div>
  </div>
</div>
`;

exports[`[ClientApp]  LoginBanner Component should match snapshot for idam login and not in comments panel 1`] = `
<div
  className="panel panel-white mt--0 mb--0 sign-in-banner"
  data-qa-sel="sign-in-banner"
  id="loginBanner"
>
  <div
    className="container"
  >
    <div
      className="LoginBanner"
      role="form"
    >
      <p
        className="display--inline"
      >
        <a
          href="/signin-url"
          title="Sign in to your NICE account"
        >
          Sign in to your NICE account
        </a>
         
        signin text
        .
         
      </p>
      <p>
        <a
          className="btn"
          href="/signin-url"
          title="Sign in to your NICE account"
        >
          Sign in
        </a>
      </p>
      <p
        className="display--inline"
      >
        <a
          href="/register-url"
          title="Register for a NICE account"
        >
          Register
        </a>
         
        for a NICE account if you don't already have one.
      </p>
    </div>
  </div>
</div>
`;
